apps/graphql/src/dataloaders/bookingsLoader/BookingsLoader.js

```diff
@@ -0,0 +1,156 @@
+// @flow
+
+import { head, last } from 'ramda';
+
+import bookings from '../../datasets/AllBookings.json';
+import type {
+  BookingApiResult,
+  Booking,
+  ApiFlight,
+  ApiRouteStop,
+  TypeSpecificData,
+} from './BookingFlowTypes';
+
+const sanitizeBookings = (
+  bookings: $ReadOnlyArray<BookingApiResult>,
+): $ReadOnlyArray<Booking> => {
+  return bookings.map(booking => {
+    const type = detectType(booking);
+    const typeSpecificData = getTypeSpecificData(booking, type);
+
+    return {
+      bid: booking.bid,
+      status: booking.status,
+      passengerCount: booking.passengers.length,
+      type,
+      ...typeSpecificData,
+    };
+  });
+};
+
+const getTypeSpecificData = (
+  booking: BookingApiResult,
+  type: $PropertyType<Booking, 'type'>,
+): TypeSpecificData => {
+  if (type === 'BookingReturn') {
+    return sanitizeReturn(booking);
+  }
+  if (type === 'BookingMulticity') {
+    sanitizeMulticity(booking);
+  }
+  return sanitizeOneWay(booking);
+};
+
+const sanitizeOneWay = (booking: BookingApiResult) => {
+  return {
+    arrival: sanitizeRouteStop(last(booking.flights).arrival),
+    departure: sanitizeRouteStop(head(booking.flights).departure),
+    segments: booking.flights.map(sanitizeFlight),
+  };
+};
+
+const sanitizeReturn = (booking: BookingApiResult) => {
+  const inboundSegments = [];
+  const outboundSegments = [];
+  const segments = booking.flights.map(flight => {
+    const segment = sanitizeFlight(flight);
+
+    if (segment.isReturn) {
+      inboundSegments.push(segment);
+    } else {
+      outboundSegments.push(segment);
+    }
+
+    return segment;
+  });
+
+  const inbound = {
+    departure: head(inboundSegments)?.departure,
```

Comment:
Why is this `?` required only here?

Reply:
If head is passed empty array, it will return undefined. I should use `?` everywhere. I will improve it.

Reply:
Fixed, I also added flow-types for ramda